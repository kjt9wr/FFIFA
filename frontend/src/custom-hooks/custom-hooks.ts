import { useEffect, useState } from "react";
import { fetchFreeAgents } from "../api/api.service";
import {
  FranchiseTagDTO,
  FreeAgentStorage,
  Player,
} from "../interfaces/interfaces";
import * as FranchiseService from "../services/franchise.service";
import { POSITION } from "../utilities/enumerations";

export const useFranchiseInfo = () => {
  const [franchiseInfo, setFranchiseInfo] = useState<FranchiseTagDTO>();

  useEffect(() => {
    const getFranchiseInformation = async () => {
      const franchiseDTO = await FranchiseService.getFranchiseTagDTO();
      setFranchiseInfo(franchiseDTO);
    };

    getFranchiseInformation();
  }, []);

  return franchiseInfo;
};

const getAvailablePlayersPosition = (
  position: string,
  playerList: Player[]
) => {
  return playerList
    .filter(
      (player: Player) =>
        player.position === position && player.rank && player.rank < 80
    )
    .sort((a: Player, b: Player) => {
      return a.rank && b.rank ? a.rank - b.rank : -1;
    });
};

export const useFreeAgents = () => {
  const [freeAgents, setFreeAgents] = useState<FreeAgentStorage>();

  useEffect(() => {
    const getFreeAgents = async () => {
      const availablePlayers = await fetchFreeAgents();

      const availableQBs = getAvailablePlayersPosition(
        POSITION.QB,
        availablePlayers.data
      );

      const availableRBs = getAvailablePlayersPosition(
        POSITION.RB,
        availablePlayers.data
      );
      const availableWRs = getAvailablePlayersPosition(
        POSITION.WR,
        availablePlayers.data
      );
      const availableTEs = getAvailablePlayersPosition(
        POSITION.TE,
        availablePlayers.data
      );

      const allFreeAgents = {
        availableQBs,
        availableRBs,
        availableWRs,
        availableTEs,
      };

      setFreeAgents(allFreeAgents);
    };

    getFreeAgents();
  }, []);

  return freeAgents;
};
