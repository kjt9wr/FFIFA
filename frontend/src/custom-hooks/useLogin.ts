import { useState } from "react";
import { useAuthContext } from "./useAuthContext";
import { fetchLogin } from "../api/api.service";

export const useLogin = () => {
  const [error, setError] = useState<any>(null);
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const { dispatch } = useAuthContext();

  const login = async (username: string, password: string) => {
    setIsLoading(true);
    setError(null);

    await fetchLogin(username, password)
      .then((response) => {
        localStorage.setItem("user", JSON.stringify(response.data));

        // update the auth context
        dispatch({ type: "LOGIN", payload: response.data });
        setIsLoading(false);
      })
      .catch((error) => {
        setIsLoading(false);
        setError(error);
      });
  };

  return { login, error, isLoading };
};
