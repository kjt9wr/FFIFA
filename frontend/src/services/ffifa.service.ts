import {
  FranchiseTagDTO,
  Owner,
  PenaltyFeeInfo,
  Player,
} from "../interfaces/interfaces";
import { BASE_CAP, getUpcomingSeasonYear } from "../utilities/constants";
import { KEEPER_CLASS_ENUM, POSITION } from "../utilities/enumerations";
import { calculateSuperMaxPrice } from "./supermax.service";

export const determineFinalPriceOfPlayer = (
  player: Player,
  franchisePrices: FranchiseTagDTO
) => {
  switch (player.keeperClass) {
    case 2:
      return getFranchisePrice(player.position, franchisePrices);
    case 3:
      return player.superMax
        ? calculateSuperMaxPrice(
            player.superMax.plan,
            player.superMax.signingYear
          )
        : -99;
    case 4:
      return getFranchisePrice(player.position, franchisePrices);
    default:
      return player.price;
  }
};

const getFranchisePrice = (playerPosition: string, prices: FranchiseTagDTO) => {
  switch (playerPosition) {
    case POSITION.QB:
      return prices.qbFranchisePrice;
    case POSITION.RB:
      return prices.rbFranchisePrice;
    case POSITION.WR:
      return prices.wrFranchisePrice;
    default:
      return prices.teFranchisePrice;
  }
};

export const pickSuperMaxOrKeeperPrice = (player: Player) => {
  return KEEPER_CLASS_ENUM.SUPERMAX === player.keeperClass && player.superMax
    ? calculateSuperMaxPrice(player.superMax.plan, player.superMax.signingYear)
    : player.price;
};

export const increaseKeeperPrice = (draftedPrice: number) => {
  return Math.max(10, Math.trunc(1.2 * draftedPrice));
};

export const calculateLuxaryPotPayout = (penaltyFees: PenaltyFeeInfo[]) => {
  const totalInPot = penaltyFees.reduce(
    (acc, owner) => acc + owner.penaltyFee,
    0
  );

  const nonOffenders = penaltyFees.filter(
    (owner) => owner.penaltyFee === 0
  ).length;

  return Math.trunc(totalInPot / nonOffenders);
};

export const calculateTotalInPot = (penaltyFees: PenaltyFeeInfo[]) => {
  return penaltyFees.reduce(
    (acc: number, owner: PenaltyFeeInfo) => acc + owner.penaltyFee,
    0
  );
};

export const calculateFollowingYearBaseCap = (
  penaltyFees: PenaltyFeeInfo[]
) => {
  const nonoffendersCount = penaltyFees.filter(
    (owner: Owner) => owner.penaltyFee === 0
  ).length;

  const offendersCount = penaltyFees.length - nonoffendersCount;

  return Math.trunc(
    BASE_CAP[getUpcomingSeasonYear()] *
      (1.05 + 0.01 * (nonoffendersCount - offendersCount))
  );
};
