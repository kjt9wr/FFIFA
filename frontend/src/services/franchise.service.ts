import { updatePlayerStatus } from "../api/api.service";
import { Player, UpdatePlayerDTO } from "../interfaces/interfaces";
import * as FFIFAService from "./ffifa.service";

export const get10MostExpensivePerPosition = (
  allKeptPlayers: Player[],
  position: string
) => {
  return allKeptPlayers
    .filter((player: Player) => player.position === position)
    .map((player: Player) => ({
      ...player,
      price: FFIFAService.pickSuperMaxOrKeeperPrice(player),
    }))
    .sort((a: Player, b: Player) => b.price - a.price)
    .slice(0, 10);
};

export const calculateFranchisePrice = (sortedPositionList: Player[]) => {
  const sum = sortedPositionList
    .slice(0, 5)
    .reduce((acc, player) => acc + player.price, 0);
  return Math.trunc(sum / Math.min(sortedPositionList.length, 5));
};

export const removeAllFranchiseTags = async (roster: Player[]) => {
  roster
    .filter((player) => player.keeperClass === 2)
    .forEach(async (player) => {
      const updateDTO: UpdatePlayerDTO = {
        keeperClass: 1,
      };
      await updatePlayerStatus(player.sleeperId, updateDTO);
    });
};

export const getFranchiseEligiblePlayers = (roster: Player[]) => {
  return roster
    .filter((player: Player) => ![3, 4].includes(player.keeperClass))
    .sort(
      (a: Player, b: Player) =>
        a.position.localeCompare(b.position) ||
        b.price - a.price ||
        a.name.localeCompare(b.name)
    );
};
