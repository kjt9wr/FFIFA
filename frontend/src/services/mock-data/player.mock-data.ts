import { Player } from "../../interfaces/interfaces";

export const MAXED_PLAYER: Player = {
  _id: "abc",
  keep: true,
  keeperClass: 3,
  name: "Najee Harris",
  owner: "",
  position: "RB",
  superMax: {
    plan: 5,
    signingYear: 2023,
  },
  price: 10,
  sleeperId: "abc",
};

export const ARBITRATED_TE: Player = {
  _id: "kittle",
  keep: true,
  keeperClass: 4,
  name: "George Kittle",
  owner: "",
  position: "TE",
  price: 50,
  sleeperId: "kittle",
};

// Kept Players

const KEPT_PLAYER = {
  keep: true,
  keeperClass: 1,
  owner: "",
};

export const MOCK_JAYDEN_DANIELS: Player = {
  ...KEPT_PLAYER,
  _id: "jayden_daniels",
  keeperClass: 2,
  name: "Jayden Daniels",
  position: "QB",
  price: 67,
  sleeperId: "jayden_daniels",
};

export const MOCK_LAMAR_JACKSON: Player = {
  ...KEPT_PLAYER,
  _id: "lamar_jackson",
  name: "Lamar Jackson",
  position: "QB",
  price: 37,
  sleeperId: "lamar_jackson",
};

export const MOCK_KYLER_MURRAY: Player = {
  ...KEPT_PLAYER,
  _id: "kyler_murray",
  name: "Kyler Murray",
  position: "QB",
  price: 37,
  sleeperId: "kyler_murray",
};

export const MOCK_PATRICK_MAHOMES: Player = {
  ...KEPT_PLAYER,
  _id: "patrick_mahomes",
  name: "Patrick Mahomes",
  position: "QB",
  price: 37,
  rank: 10,
  sleeperId: "patrick_mahomes",
};

export const MOCK_JALEN_HURTS: Player = {
  ...KEPT_PLAYER,
  _id: "jalen_hurts",
  name: "Jalen Hurts",
  position: "QB",
  price: 37,
  sleeperId: "jalen_hurts",
};

export const MOCK_BIJAN_ROBINSON: Player = {
  ...KEPT_PLAYER,
  _id: "bijan_robison",
  keeperClass: 2,
  name: "Bijan Robinson",
  position: "RB",
  price: 121,
  sleeperId: "bijan_robison",
};

export const MOCK_SAQUON_BARKLEY: Player = {
  ...KEPT_PLAYER,
  _id: "saquon_barkley",
  keeperClass: 2,
  name: "Saquon Barkley",
  position: "RB",
  price: 100,
  sleeperId: "saquon_barkley",
};

export const MOCK_JAHMYR_GIBBS: Player = {
  ...KEPT_PLAYER,
  _id: "jahmyr_gibbs",
  keeperClass: 2,
  name: "Jahmyr Gibbs",
  position: "RB",
  price: 98,
  sleeperId: "jahmyr_gibbs",
};

export const MOCK_JOSH_JACOBS: Player = {
  ...KEPT_PLAYER,
  ...KEPT_PLAYER,
  _id: "josh_jacobs",
  name: "Josh Jacobs",
  position: "RB",
  price: 81,
  rank: 12,
  sleeperId: "josh_jacobs",
};

export const MOCK_DERRICK_HENRY: Player = {
  ...KEPT_PLAYER,
  _id: "derrick_henry",
  name: "Derrick Henry",
  position: "RB",
  price: 79,
  firstKeepYear: 2023,
  rank: 6,
  sleeperId: "derrick_henry",
};

export const MOCK_NAJEE_HARRIS: Player = {
  ...KEPT_PLAYER,
  _id: "najee_harris",
  keeperClass: 3,
  name: "Najee Harris",
  sleeperId: "najee_harris",
  position: "RB",
  owner: "Patrick",
  superMax: {
    plan: 5,
    signingYear: 2022,
  },
  price: 130,
};

export const MOCK_BROCK_BOWERS: Player = {
  ...KEPT_PLAYER,
  _id: "brock_bowers",
  keeperClass: 2,
  name: "Brock Bowers",
  position: "TE",
  price: 37,
  sleeperId: "brock_bowers",
};

export const MOCK_SAM_LAPORTA: Player = {
  ...KEPT_PLAYER,
  _id: "sam_laporta",
  name: "Sam Laporta",
  position: "TE",
  price: 12,
  rank: 4,
  sleeperId: "sam_laporta",
};

export const MOCK_DRAKE_LONDON: Player = {
  ...KEPT_PLAYER,
  _id: "drake_london",
  name: "Drake London",
  position: "WR",
  price: 64,
  keeperClass: 2,
  sleeperId: "drake_london",
};

export const MOCK_AJ_BROWN: Player = {
  ...KEPT_PLAYER,
  _id: "aj_brown",
  name: "AJ Brown",
  position: "WR",
  price: 61,
  keeperClass: 2,
  sleeperId: "aj_brown",
};

export const MOCK_CEEDEE_LAMB: Player = {
  ...KEPT_PLAYER,
  _id: "ceedee_lamb",
  name: "CeeDee Lamb",
  position: "WR",
  price: 56,
  sleeperId: "ceedee_lamb",
};

export const MOCK_TYREEK_HILL: Player = {
  ...KEPT_PLAYER,
  _id: "tyreek_hill",
  name: "Tyreek Hill",
  position: "WR",
  price: 56,
  rank: 10,
  sleeperId: "tyreek_hill",
};

export const MOCK_JAMARR_CHASE: Player = {
  ...KEPT_PLAYER,
  _id: "jamarr_chase",
  name: "Jamarr Chase",
  position: "WR",
  price: 52,
  firstKeepYear: 2022,
  sleeperId: "jamarr_chase",
};

export const MOCK_MALIK_NABERS: Player = {
  ...KEPT_PLAYER,
  _id: "malik_nabers",
  name: "Malik Nabers",
  position: "WR",
  price: 51,
  sleeperId: "malik_nabers",
};

export const MOCK_JALEN_COKER: Player = {
  ...KEPT_PLAYER,
  _id: "jalen_coker",
  name: "Jalen Coker",
  position: "WR",
  price: 10,
  rank: 30,
  sleeperId: "jalen_coker",
};

export const MOCK_CHRIS_OLAVE: Player = {
  _id: "chris_olave",
  keep: false,
  keeperClass: 1,
  name: "Chris Olave",
  owner: "",
  position: "WR",
  price: 62,
  sleeperId: "chris_olave",
};

export const MOCKED_ROSTER = [
  MOCK_AJ_BROWN,
  MOCK_CEEDEE_LAMB,
  MOCK_JALEN_HURTS,
  MOCK_CHRIS_OLAVE,
];

export const MOCKED_ROSTER_WITH_SUPERMAX = [
  ...MOCKED_ROSTER,
  MOCK_NAJEE_HARRIS,
];
