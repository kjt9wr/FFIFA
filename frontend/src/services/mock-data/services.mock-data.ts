import { FreeAgentStorage, Player } from "../../interfaces/interfaces";
import {
  MAXED_PLAYER,
  MOCK_AJ_BROWN,
  MOCK_BIJAN_ROBINSON,
  MOCK_BROCK_BOWERS,
  MOCK_CEEDEE_LAMB,
  MOCK_DERRICK_HENRY,
  MOCK_DRAKE_LONDON,
  MOCK_JAHMYR_GIBBS,
  MOCK_JALEN_COKER,
  MOCK_JALEN_HURTS,
  MOCK_JAMARR_CHASE,
  MOCK_JAYDEN_DANIELS,
  MOCK_JOSH_JACOBS,
  MOCK_KYLER_MURRAY,
  MOCK_LAMAR_JACKSON,
  MOCK_MALIK_NABERS,
  MOCK_NAJEE_HARRIS,
  MOCK_PATRICK_MAHOMES,
  MOCK_SAM_LAPORTA,
  MOCK_SAQUON_BARKLEY,
  MOCK_TYREEK_HILL,
} from "./player.mock-data";

export const FRANCHISE_TAG_SAMPLE_DATA = {
  qbFranchisePrice: 25,
  keptQBs: [],
  rbFranchisePrice: 75,
  keptRBs: [MOCK_DERRICK_HENRY, MAXED_PLAYER],
  wrFranchisePrice: 50,
  keptWRs: [],
  teFranchisePrice: 30,
  keptTEs: [],
};

export const MOCKED_PENALTY_FEES = [
  { name: "Kevin", penaltyFee: 20 },
  { name: "Justin", penaltyFee: 50 },
  { name: "Matt", penaltyFee: 0 },
  { name: "Luigi", penaltyFee: 0 },
  { name: "Alex", penaltyFee: 30 },
];

export const KEPT_PLAYERS: Player[] = [
  MOCK_JAYDEN_DANIELS,
  MOCK_JALEN_HURTS,
  MOCK_KYLER_MURRAY,
  MOCK_LAMAR_JACKSON,
  MOCK_PATRICK_MAHOMES,
  MOCK_BIJAN_ROBINSON,
  MOCK_SAQUON_BARKLEY,
  MOCK_JAHMYR_GIBBS,
  MOCK_JOSH_JACOBS,
  MOCK_DERRICK_HENRY,
  MOCK_NAJEE_HARRIS,
  MOCK_BROCK_BOWERS,
  MOCK_SAM_LAPORTA,
  MOCK_AJ_BROWN,
  MOCK_DRAKE_LONDON,
  MOCK_CEEDEE_LAMB,
  MOCK_TYREEK_HILL,
  MOCK_JAMARR_CHASE,
  MOCK_MALIK_NABERS,
  MOCK_JALEN_COKER,
];

export const MOCK_FREE_AGENTS: FreeAgentStorage = {
  availableQBs: [MOCK_PATRICK_MAHOMES],
  availableRBs: [MOCK_DERRICK_HENRY, MOCK_JOSH_JACOBS],
  availableWRs: [MOCK_TYREEK_HILL, MOCK_JALEN_COKER],
  availableTEs: [MOCK_SAM_LAPORTA],
};
