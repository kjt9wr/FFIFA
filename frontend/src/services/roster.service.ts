import { fetchSingleOwner } from "../api/api.service";
import { Player, FranchiseTagDTO } from "../interfaces/interfaces";
import { getUpcomingYearIndex } from "../utilities/constants";
import { determineFinalPriceOfPlayer } from "./ffifa.service";

export const calculateLuxaryTaxLine = (cap: number) => Math.trunc(cap * 0.55);

export const calculatePenaltyFee = (
  roster: Player[],
  franchisePrices: FranchiseTagDTO,
  maxCap: number
) => {
  const penaltyFee =
    calculateTotalKeeperPrice(roster, franchisePrices) -
    calculateLuxaryTaxLine(maxCap);
  return penaltyFee > 0 ? penaltyFee : 0;
};

export const calculateTotalKeeperPrice = (
  roster: Player[],
  franchisePrices: FranchiseTagDTO
) => {
  return roster
    .filter((keptPlayer) => keptPlayer.keep)
    .reduce(
      (acc, player) =>
        acc + determineFinalPriceOfPlayer(player, franchisePrices),
      0
    );
};

export const getOwnersCap = async (name: string) => {
  const owner = await fetchSingleOwner(name);
  console.log(owner.data);
  return owner.data[0].cap[getUpcomingYearIndex()];
};
