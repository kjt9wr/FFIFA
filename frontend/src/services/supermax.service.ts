import { BASE_CAP, getUpcomingSeasonYear } from "../utilities/constants";

const THREE_YEAR_RATES = [0.25, 0.2375, 0.2256];
const FOUR_YEAR_RATES = [0.2375, 0.2256, 0.2143, 2036];
const FIVE_YEAR_RATES = [0.225, 0.2138, 0.2031, 0.1929, 0.1833];
const upcomingYear = getUpcomingSeasonYear();
export const getCurrentSuperMaxYear = (signingYear: number) => {
  return Number(upcomingYear) - signingYear;
};

export const calculateSuperMaxPrice = (
  durationInYears: number,
  signingYear: number
) => {
  const currentYear = getCurrentSuperMaxYear(signingYear);

  switch (durationInYears) {
    case 3:
      return Math.trunc(
        BASE_CAP[upcomingYear] * THREE_YEAR_RATES[currentYear - 1]
      );
    case 4:
      return Math.trunc(
        BASE_CAP[upcomingYear] * FOUR_YEAR_RATES[currentYear - 1]
      );
    case 5:
      return Math.trunc(
        BASE_CAP[upcomingYear] * FIVE_YEAR_RATES[currentYear - 1]
      );
    default:
      return 0;
  }
};
