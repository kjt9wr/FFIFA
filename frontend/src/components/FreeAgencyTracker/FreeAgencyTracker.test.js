import { render, screen } from "@testing-library/react";
import * as customHooks from "../../custom-hooks/custom-hooks";
import { MOCK_FREE_AGENTS } from "../../services/mock-data/services.mock-data";
import FreeAgency from "./FreeAgencyTracker";

describe("free agency page", () => {
  it("renders successfully", async () => {
    jest.spyOn(customHooks, "useFreeAgents").mockReturnValue(MOCK_FREE_AGENTS);

    const view = render(<FreeAgency />);

    expect(await screen.findByText("Derrick Henry")).toBeTruthy();
    expect(view).toMatchSnapshot();
  });
});
