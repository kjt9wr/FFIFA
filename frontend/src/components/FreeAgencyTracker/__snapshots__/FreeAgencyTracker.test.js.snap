// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`free agency page renders successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="text-center"
        >
           Free Agents 
        </h2>
        <div
          class="accordion"
        >
          <div
            class="accordion-item"
            style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
          >
            <h2
              class="accordion-header"
            >
              <button
                class="accordion-button"
                type="button"
              >
                QB
              </button>
            </h2>
            <div
              class="accordion-collapse collapse show"
            >
              <div
                class="accordion-body"
              >
                <table
                  class="table table-sm table-borderless"
                >
                  <tbody>
                    <tr>
                      <td
                        style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
                      >
                         
                        Patrick Mahomes
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <div
            class="accordion-item"
            style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
          >
            <h2
              class="accordion-header"
            >
              <button
                class="accordion-button collapsed"
                type="button"
              >
                RB
              </button>
            </h2>
            <div
              class="accordion-collapse collapse"
            >
              <div
                class="accordion-body"
              >
                <table
                  class="table table-sm table-borderless"
                >
                  <tbody>
                    <tr>
                      <td
                        style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
                      >
                         
                        Derrick Henry
                      </td>
                    </tr>
                    <tr>
                      <td
                        style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
                      >
                         
                        Josh Jacobs
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <div
            class="accordion-item"
            style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
          >
            <h2
              class="accordion-header"
            >
              <button
                class="accordion-button collapsed"
                type="button"
              >
                WR
              </button>
            </h2>
            <div
              class="accordion-collapse collapse"
            >
              <div
                class="accordion-body"
              >
                <table
                  class="table table-sm table-borderless"
                >
                  <tbody>
                    <tr>
                      <td
                        style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
                      >
                         
                        Tyreek Hill
                      </td>
                    </tr>
                    <tr>
                      <td
                        style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
                      >
                         
                        Jalen Coker
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <div
            class="accordion-item"
            style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
          >
            <h2
              class="accordion-header"
            >
              <button
                class="accordion-button collapsed"
                type="button"
              >
                TE
              </button>
            </h2>
            <div
              class="accordion-collapse collapse"
            >
              <div
                class="accordion-body"
              >
                <table
                  class="table table-sm table-borderless"
                >
                  <tbody>
                    <tr>
                      <td
                        style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
                      >
                         
                        Sam Laporta
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="text-center"
      >
         Free Agents 
      </h2>
      <div
        class="accordion"
      >
        <div
          class="accordion-item"
          style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
        >
          <h2
            class="accordion-header"
          >
            <button
              class="accordion-button"
              type="button"
            >
              QB
            </button>
          </h2>
          <div
            class="accordion-collapse collapse show"
          >
            <div
              class="accordion-body"
            >
              <table
                class="table table-sm table-borderless"
              >
                <tbody>
                  <tr>
                    <td
                      style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
                    >
                       
                      Patrick Mahomes
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          class="accordion-item"
          style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
        >
          <h2
            class="accordion-header"
          >
            <button
              class="accordion-button collapsed"
              type="button"
            >
              RB
            </button>
          </h2>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="accordion-body"
            >
              <table
                class="table table-sm table-borderless"
              >
                <tbody>
                  <tr>
                    <td
                      style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
                    >
                       
                      Derrick Henry
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
                    >
                       
                      Josh Jacobs
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          class="accordion-item"
          style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
        >
          <h2
            class="accordion-header"
          >
            <button
              class="accordion-button collapsed"
              type="button"
            >
              WR
            </button>
          </h2>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="accordion-body"
            >
              <table
                class="table table-sm table-borderless"
              >
                <tbody>
                  <tr>
                    <td
                      style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
                    >
                       
                      Tyreek Hill
                    </td>
                  </tr>
                  <tr>
                    <td
                      style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
                    >
                       
                      Jalen Coker
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          class="accordion-item"
          style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
        >
          <h2
            class="accordion-header"
          >
            <button
              class="accordion-button collapsed"
              type="button"
            >
              TE
            </button>
          </h2>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="accordion-body"
            >
              <table
                class="table table-sm table-borderless"
              >
                <tbody>
                  <tr>
                    <td
                      style="background-color: rgb(69, 73, 94); color: rgb(248, 248, 242);"
                    >
                       
                      Sam Laporta
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
